src/app/lib/views/browser/list.js

```diff
@@ -229,87 +192,98 @@
-                $('#loading-more-animi').hide();
-                $('.status-loadmore').show();
+                var loadmore = $(document.getElementById('load-more-item'));
+                loadmore.children('.status-loadmore').css('display', 'block');
+                loadmore.children('.loading-container').css('display', 'none');
             }
 
-            $('.filter-bar').on('mousedown', function (e) {
-                if (e.target.localName !== 'div') {
-                    return;
-                }
-                _.defer(function () {
-                    self.$('.items:first').focus();
-                });
-            });
-            $('.items').attr('tabindex', '1');
-            _.defer(function () {
-                self.checkFetchMore();
-                self.$('.items:first').focus();
-            });
-
         },
 
         checkFetchMore: function () {
-            // if load more is visible onLoaded, fetch more results
-            if (elementInViewport(this.$el, $('#load-more-item'))) {
-                this.collection.fetchMore();
+            var loadmore = $(document.getElementById('load-more-item'));
+
+            return ( // if load more is visible onLoaded, fetch more results
+                loadmore.is(':visible') 
+                && Common.isElementInViewport(loadmore)
+            ) ? this.collection.fetchMore() : false;
+        },
+
+        itemsPerRow: function () {
+            var total = 0;
+            var items = $(document.querySelector('.items')).children('.item');
+
+            for (var i = 0; i < items.length; i++) {
+                var el = $(items[i]);
+
+                if (el.prev().length > 0 && el.position().top !== el.prev().position().top) {
+                    break;
+                }
+                
+                total++; 
             }
+
+            return total;
         },
 
         completerow: function () {
-            var elms = this.addloadmore();
-            elms += this.addghosts();
+            NUM_ITEMS_IN_ROW = this.itemsPerRow();
 
-            $('.ghost, #load-more-item').remove();
-            $('.items').append(elms);
+            var items = $(document.querySelector('.items'));
 
-            this.showloadmore();
-        },
+            var loadmore = 
+                '<div id="load-more-item" class="load-more">' +
+                    '<span class="status-loadmore">' + 
+                        i18n.__('Load More') + 
+                    '</span>' +
+                    '<div class="loading-container">' +
+                        '<div class="sk-folding-cube">' +
+                              '<div class="sk-cube1 sk-cube"></div>' +
+                              '<div class="sk-cube2 sk-cube"></div>' +
+                              '<div class="sk-cube4 sk-cube"></div>' +
+                              '<div class="sk-cube3 sk-cube"></div>' +
+                        '</div>' +
+                    '</div>' +
+                '</div>';
```

Comment:
we can put it in a template and do the loading ourselves. really too much js, and it's already in list.tpl so we need to keep both in sync.